### DISCUSSION TITLE
same log4j version for ejbca and wildfly

### DISCUSSION THREAD

**Discussion by hanuamantha** [Keyfactor/ejbca-ce#584] [Q&A] [answered]
Hello Everyone, 

I am new to EJBCA. I understand that Ejbca is deployed on wildfly application server. When I search for log4j version in my ejbca container I found below results.
```
/var/lib/**wildfly**/modules/system/layers/base/org/apache/logging/log4j/api/main/log4j-api-2.17.2.jar
/opt/**ejbca**/dist/ejbca-ws-cli/lib/log4j-1.2-api-2.17.1.jar
/opt/**ejbca**/dist/ejbca-ws-cli/lib/log4j-core-2.17.1.jar
/opt/**ejbca**/dist/ejbca-ws-cli/lib/log4j-api-2.17.1.jar
```

We are generating jar files using source code. In wildfly-core pom.xml I can see that
`        <version.org.apache.logging.log4j>2.17.2</version.org.apache.logging.log4j>
`In ejbca when I search for log4j jars, found below output
```
./lib/log4j-1.2-api-2.17.1.jar
./lib/log4j-api-2.17.1.jar
./lib/log4j-core-2.17.1.jar
```

Can I remove` log4j-1.2-api-2.17.1.jar log4j-api-2.17.1.jar log4j-core-2.17.1.jar` in ejbca so that it will use wildfly log4j version i.e.,` log4j-api-2.17.2`?

**Comment by primetomas** [comment, marked as answer]
There is no need to do anything. If you look inside the ejbca.ear file that is what is deployed to WildFly you will see that there are no log4j jars in there.  So only the WildFly log4j files are used in runtime by the server.

The jars in EJBCA is there for compilation, and for standalone web services clients. What you point to, ejbca-ws-cli, is a standalone command line client (look for clientToolBox in the documentation) that has no interaction with WildFly at all, so yes the command line client need these jars on it's own.

**Reply by hanuamantha** [reply]
Thank you
